### PR TITLE
Implement Settings UI and Hierarchical File Structure

### PR DESCRIPTION
**Pull request by bencole12345**
Two features: there is now an interface for building a configuration file for the radar devices, and the app now supports a proper hierarchical directory structure for storing data files.

### CODE REVIEW

**Comment by bencole12345** [comment]
**I just spotted a bug**: I haven't fixed the action bar, so the sign in and upload buttons appear no matter what screen you're looking at. I'll fix this tomorrow morning.

File: app/src/main/java/uk/ac/cam/cl/juliet/data/InternalDataHandler.java [hunk 1]
```diff
@@ -14,23 +14,26 @@
...
-            rootEmpty = (root.listFiles() == null);
-            if (listeners == null) {
-                listeners = new ArrayList<>();
-            }
         }
         return INSTANCE;
     }
 
+    public InternalDataHandler() {
+        root = new File(Environment.getExternalStorageDirectory().getAbsolutePath(), ROOT_NAME);
```
Comment (patricoferris) [inline app/src/main/java/uk/ac/cam/cl/juliet/data/InternalDataHandler.java:30]: Wait why are you creating a public constructor for this class? I was trying to make it a singleton?

**Review by patricoferris — COMMENTED** [review]
Had a bit of a look over this and I don't see anything that stands out but I'll need to test actually on the emulator tomorrow - it will undoubtedly break the live processing I'm almost sure of it but that's okay... Other than that great stuff.

File: app/src/main/java/uk/ac/cam/cl/juliet/data/InternalDataHandler.java (hunk 1 above)
Comment (bencole12345) [reply app/src/main/java/uk/ac/cam/cl/juliet/data/InternalDataHandler.java:30]: Oops, that's a typo - good spot!

**Review by patricoferris — APPROVED** [review]
LGTM and what I saw today looked good - awesome work

**Pull request merged**